app/extlinux_control.py

```diff
@@ -88,6 +98,42 @@ def get_dev_by_partlabel(cls, partlabel: str) -> str:
     def get_dev_by_partuuid(cls, partuuid: str) -> str:
         return cls._findfs("PARTUUID", partuuid)
 
+    @classmethod
+    def get_rootfs_dev(cls) -> str:
+        return cls._findmnt("/ -o SOURCE -n")
+
+    @classmethod
+    def get_parent_dev(cls, child_device: str) -> str:
+        """
+        When `/dev/nvme0n1p1` is specified as child_device, /dev/nvme0n1 is returned.
+        """
+        cmd = f"-ipn -o PKNAME {child_device}"
+        return cls._lsblk(cmd)
+
+    @classmethod
+    def get_family_devs(cls, parent_device: str) -> list:
+        """
+        When `/dev/nvme0n1` is specified as parent_device,
+        ["NAME=/dev/nvme0n1", "NAME=/dev/nvme0n1p1", "NAME=/dev/nvme0n1p2"] is returned.
+        """
+        cmd = f"-Pp -o NAME {parent_device}"
+        return cls._lsblk(cmd).split("\n")
+
+    @classmethod
+    def get_sibling_dev(cls, device: str) -> str:
+        """
+        When `/dev/nvme0n1p1` is specified as child_device, /dev/nvme0n1p2 is returned.
+        """
+        parent = cls.get_parent_dev(device)
+        family = cls.get_family_devs(parent)
+        pa = re.compile(r'NAME="(?P<dev>.*)"')
+        for blk in family:
+            m = pa.match(blk)
+            dev = m.group("dev")
+            if dev != parent and dev != device:
+                return dev
+        raise OtaErrorUnrecoverable(f"lsblk output={family} is illegal")
```

Comment:
No need to use regex here.
Also handling unexpected partition layout here, i.e., the device doesn't have exact 2 partitions.